source/opt/function.cpp

```diff
@@ -193,6 +193,30 @@ BasicBlock* Function::InsertBasicBlockBefore(
   return nullptr;
 }
 
+uint32_t Function::GetReturnValueId() const {
+  auto post_dominator_analysis =
+      blocks_.front()->GetLabel()->context()->GetPostDominatorAnalysis(this);
+  for (auto& block : blocks_) {
+    if (post_dominator_analysis->Dominates(block.get(), entry().get()) &&
+        block->tail()->opcode() == SpvOpReturnValue) {
+      return block->tail()->GetSingleWordOperand(0);
+    }
+  }
+  return 0;
+}
```

Comment:
Please move this into the transformation class.  It only makes sense in the case where there is a single return instruction in the function.  That's something the transformation assumes, but not something that is true in general, so the function doesn't belong in function.cpp.

Reply:
Done.